simplerisk/includes/functions.php

```diff
@@ -56,13 +56,19 @@ function db_open()
     // Connect to the database
     try
     {
-        $options = array(
-            PDO::MYSQL_ATTR_SSL_KEY => DB_SSL_CERTIFICATE_PATH
-        );
+        if(DB_SSL_CERTIFICATE_PATH != '') {
+            $options = array(
+                PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
+                PDO::MYSQL_ATTR_SSL_CA => DB_SSL_CERTIFICATE_PATH
+            );
+        } else {
+            $options = array(
+                PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION
+            );
+        }
```

Comment:
I haven't tested this yet, but this would not work for existing SimpleRisk users who do not have the DB_SSL_CERTIFICATE_PATH defined and we do not typically overwrite or add to the config.php file with upgrades.  I think what you're looking for here is something like "if (defined(DB_SSL_CERTIFICATE_PATH))" .  That would check if the constant is defined and then use that.

Reply:
Come to think of it, we probably need to check for both whether it is defined and not an empty string.

Reply:
I'm working on integrating this now.  Going to do it slightly differently, but your code here is helpful.

Reply:
OK, here's what I added into our release code (should go live at the end of the month):

        // Set the default options array
        $options = array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION);

        // If a database SSL certificate path has been defined
        if (defined("DB_SSL_CERTIFICATE_PATH") && DB_SSL_CERTIFICATE_PATH != '')
        {
                // Add the SSL certificate to the options array
                $options[PDO::MYSQL_ATTR_SSL_CA] = DB_SSL_CERTIFICATE_PATH;
        }

        $GLOBALS['db'] = new PDO("mysql:charset=UTF8;dbname=".DB_DATABASE.";host=".DB_HOSTNAME.";port=".DB_PORT,DB_USERNAME,DB_PASSWORD, $options);

Reply:
I've pulled those changes into my fork, and thus far all seems to be good. Let me know when it's released and I'll close down my fork. 👏 🙏